File: frontend/src/app/service/lesson.service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {API} from "../config/config";
import {LessonDTO} from "../domain/dto/lesson.dto";

@Injectable({
  providedIn: 'root'
})
export class LessonService {

  private http = inject(HttpClient)

  find(
    id: number
  ): Observable<any> {
    return this.http
      .get(`${API}/lesson/` + id)
  }

  delete(
    id: number
  ): Observable<any> {
    return this.http
      .delete(`${API}/lesson/` + id)
  }

  update(
    id: number,
    lessonDTO: LessonDTO
  ): Observable<any> {
    return this.http
      .patch(`${API}/lesson/` + id, lessonDTO);
  }


}

File: backend/course-service/src/main/java/danekerscode/courseservice/service/LessonService.java
package danekerscode.courseservice.service;

import danekerscode.courseservice.dto.LessonDTO;
import danekerscode.courseservice.mapper.LessonMapper;
import danekerscode.courseservice.model.Course;
import danekerscode.courseservice.model.Lesson;

public interface LessonService {

    void delete(Long id);

    Lesson save(LessonDTO dto, Course course);

    void update(Long id, LessonDTO dto);

    Lesson findById(Long id);

}
